proposals/0029-cooperative-vector.md

```diff
@@ -102,383 +111,335 @@ Introduce new HLSL intrinsics to accelarate matrix-vector operations. In this sp
 
 ## Detailed design
 
-### Intrinsics for Vector-Matrix Operations
+### Matrix-Vector Multiply and Multiply-Add Operations
 
-**Matrix-Vector Multiply and Add Intrinsic**
-
-Intrinsics for specifying a multiplication operation between a matrix(Dim: M * K) loaded from memory and aa vector(Dim: K), a
-variant of this with an add, where a bias vector(Dim: K), loaded from memory, is added to the result vector(Dim: M) of the matrix-vector
-multiply operation.
-
-Note that the dimensions of the matrix are `M X K` versus `M x N`  usually found in linear algebra texbooks. This is to
-futureproof for potential Matrix-Matrix operations in the future where the inputs could be `M X K` and `K x N` to
-produce an `M X N` result matrix.
+#### Syntax
+ 
+> NOTES FOR REVIEW, REMOVE BEFORE MERGING: 
+>  
+> Added @dx.op.matvecmuladd version, and also renamed  from
+> @dx.op.vecmatmul[add] to @dx.op.matvecmul[add] since it is a left matrix
+> multiply.
 
-The `InputVector` is an HLSL vector and the `Matrix` and `BiasVector` are loaded from memory at specified offsets.
+``` llvm 
+declare <[NUMo] x [TYo] @dx.op.matvecmul.v[NUMo][TYo].v[NUMi][TYi](
+    i32               ; opcode
+    <[NUMi] x [TYi]>, ; input vector
+    i32,              ; input interpretation
+    %dx.types.Handle, ; matrix resource
+    i32,              ; matrix offset
+    i32,              ; matrix interpretation
+    i32,              ; matrix M dimension    
+    i32,              ; matrix K dimension    
+    i32,              ; matrix layout
+    i32,              ; matrix transpose      <<< should this be i1?
```

Comment:
The matrix transpose argument is intended to be a bool value.  Should this be `i1`?

Reply:
Yep, makes sense.